File: src/hash/lc49.go
package hash

import (
	"slices"
	"strings"
)

func groupAnagrams(strs []string) [][]string {
	s2ss := make(map[string][]string)

	for _, str := range strs {
		ss := strings.Split(str, "")
		slices.Sort(ss)
		s := strings.Join(ss, "")
		s2ss[s] = append(s2ss[s], str)
	}

	ans := make([][]string, 0)
	for _, ss := range s2ss {
		ans = append(ans, ss)
	}
	return ans
}

var GroupAnagrams = groupAnagrams

File: src/hash/lc49.ts
function groupAnagrams(strs: string[]): string[][] {
  const s2ss: Record<string, string[]> = {};
  for (const str of strs) {
    const ss = str.split("");
    ss.sort();
    const s = ss.join("");
    if (!s2ss[s]) {
      s2ss[s] = [];
    }
    s2ss[s].push(str);
  }
  const ans: string[][] = [];
  for (const key in s2ss) {
    ans.push(s2ss[key]);
  }
  return ans;
}

export default groupAnagrams;
